cpp/modules/deck.gl/core/src/lib/deck.h

```diff
@@ -38,9 +39,12 @@ class Deck : public Component {
   class Props;
   auto props() { return std::dynamic_pointer_cast<Props>(this->_props); }
   void setProps(std::shared_ptr<Deck::Props> props);
+  void setProps(std::shared_ptr<Deck::Props> props, probegl::Error* error);
 
   struct RenderingOptions;
   explicit Deck(std::shared_ptr<Deck::Props> props = std::make_shared<Deck::Props>());
+  static auto make(probegl::Error* error, std::shared_ptr<Deck::Props> props = std::make_shared<Deck::Props>())
```

Comment:
Total Nit: I kind of like the constructor and desctructor together so maybe put the `make_shared` first?